### PR TITLE
Feat/datanode union

### PR DESCRIPTION
**Pull request by paulineribeyre**
- the fields available in the datanode interface = the union of all the data nodes' fields.
- i also changed the way the fields are created, to fix a bug where integer fields were displayed as strings
- for now the links are not included

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
Jenkins Build 3 : time taken 1 hr 0 min

File: peregrine/resources/submission/graphql/node.py [hunk 1]
```diff
@@ -830,27 +831,49 @@ def get_fields():
...
-            set(schema['properties'].keys())
-            for schema in dictionary.schema.values()
-            if schema['category'].endswith('_file')
-        ]
+        # union of all the data nodes' possible fields
+        shared_fields_dict = {
+            'id': graphene.String(),
+            'type': graphene.String()
+        }
+        for cls in get_data_subclasses(): # select the data nodes
```
Comment (rudyardrichter) [inline peregrine/resources/submission/graphql/node.py]: nitpick: can you use `subclass` so it's not a python keyword?

File: peregrine/resources/submission/graphql/node.py [hunk 2]
```diff
@@ -830,27 +831,49 @@ def get_fields():
...
     if not DataNode.shared_fields:
 
-        # fields lists the set of node fields, for every data node in the dictionary schema (nodes ending with '_file')
-        fields = [
-            set(schema['properties'].keys())
-            for schema in dictionary.schema.values()
-            if schema['category'].endswith('_file')
-        ]
+        # union of all the data nodes' possible fields
+        shared_fields_dict = {
```
Comment (rudyardrichter) [inline peregrine/resources/submission/graphql/node.py]: am I correct that this snippet would be equivalent?
```python
shared_fields_dict = {
    k: lookup_graphql_type(types[0])()
    for subclass in get_data_subclasses()
    for k, types in subclass.__pg_properties__.items()
    if k not in subclass._pg_edges.keys()
}
shared_fields_dict.update({
    'id': graphene.String(),
    'type': graphene.String(),
})
```

File: peregrine/resources/submission/graphql/node.py [hunk 3]
```diff
@@ -830,27 +831,49 @@ def get_fields():
...
+    data_subclasses = None
     shared_fields = None # fields shared by all data nodes in the dictionary
 
 
+def get_data_subclasses():
+    """Return a list of the subclasses representing data categories."""
+
+    if not DataNode.data_subclasses:
+        # get the list of categories that are data categories (end with _file)
+        data_subclasses_labels = [
```
Comment (rudyardrichter) [inline peregrine/resources/submission/graphql/node.py]: can be set not list

**Comment by PlanXCyborg** [comment]
Jenkins Build 4 : time taken 26 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 5 : time taken 52 sec

**Pull request merged**